src/main/java/org/snomed/snowstorm/core/data/services/ConceptService.java

```diff
@@ -153,55 +158,62 @@ public Concept find(String id, List<LanguageDialect> languageDialects, BranchTim
 	}
 
 	public ConceptHistory loadConceptHistory(String conceptId, List<CodeSystemVersion> codeSystemVersions, boolean showFutureVersions) {
+		int now = Integer.parseInt(DATE_TIME_FORMATTER.format(Instant.now()));
 		Map<String, BranchCriteria> branchCriteria = new HashMap<>();
 		for (CodeSystemVersion codeSystemVersion : codeSystemVersions) {
-			String branchPath = codeSystemVersion.getBranchPath();
-			branchCriteria.put(branchPath, versionControlHelper.getBranchCriteria(branchPath));
+			if (showFutureVersions || codeSystemVersion.getEffectiveDate() < now) {
+				String branchPath = codeSystemVersion.getBranchPath();
+				branchCriteria.put(branchPath, versionControlHelper.getBranchCriteria(branchPath));
+			}
 		}
 
-		Function<String, BoolQueryBuilder> defaultBoolQueryFunction = cId -> {
-			BoolQueryBuilder someReleaseBranch = boolQuery();
-			BoolQueryBuilder boolQueryBuilder = boolQuery();
+		BiFunction<String, ComponentType, BoolQueryBuilder> defaultFullQuery = (cId, cT) -> {
+			BoolQueryBuilder fullQuery = boolQuery();
+			fullQuery.should(
```

Comment:
This should be a `must` ?

Reply:
Yep, good catch. Changed it and re-tested; all working as expected.